### ISSUE TITLE
Dashboard Mails

### ISSUE THREAD

**Issue by ghost** [Z3r0byte/Magistify#13]
Zodra er Magister-mails in de app zitten, kun je deze ook toevoegen aan het dashboard.

**Comment by ghost** [comment]
@Z3r0byte 
Ik zie dat je hulp nodig hebt, ik heb het volgende ontdekt (als je dat zelf nog niet gezien had):

Eerst stuur je een verzoek naar 
>/api/personen/**id**/berichten/mappen/

Vervolgens krijg je JSON terug waarvan je *Items* nodig hebt. 
Dan ga je door alle mappen heen en haal je voor ieder alle berichten op, door: 
> /api/personen/**id**/berichten?mapId=**getal**&orderby=soort+DESC&skip=0&top=25

**het 'getal' start altijd bij 0 (net zoals bij een normale iteration, maar ik dacht ik vermeld het nog even)**

**Ik heb hier de variabele die je als json terug krijgt even "naam" genoemd en het iteration-getal (normaal gesproken *i*) heb ik "getal" genoemd, sorry voor de eventuele verwarring.**

Vervolgens krijg je weer een JSON terug en daarvan ga je per e-mail dit doen om ze te tonen:
> *naam*.Items[**getal**].Afzender.Naam

> *naam*.Items[**getal**].Onderwerp

En als de mail dan aangeklikt wordt stuur je dit verzoek:

> /api/personen/**id**/berichten/***naam*.Items[*getal*].Id?berichtSoort=**

Maar hierachter moet nog een getal komen te staan, namelijk welk type bericht dit is, de enige tijdelijke oplossing hiervoor is gewoon alles proberen van 0 tot 9, totdat je niet
`{"Message":"Het opgegeven berichtsoort is ongeldig."}`
als antwoord krijgt.

Ook moet je controleren of de mail niet door jezelf verstuurd is, dit doe je door dit te checken:
> *naam*.Items["**getal**"].Afzender.Id === **leerling-id**

En je kunt nog tonen welke ontvangers er zijn (hiervoor is er een tweede loop nodig):
> *naam*.Items["**getal**"].Ontvangers["**getal2**"].Naam


Hopelijk begreep je iets van wat ik je geprobeerd heb uit te leggen, anders hoor ik graag wat er nog onduidelijk is.

**Comment by Z3r0byte** [comment]
Dankje, de reden dat ik hulp nodig heb, is omdat ze bij ons op school berichten hebben uitgeschakeld.

**Comment by ghost** [comment]
@Z3r0byte Oh dat is inderdaad erg lastig, hopelijk heb je iets aan mijn informatie en ik wil graag de berichten functie voor je testen

**Comment by ghost** [comment]
@Z3r0byte 
Bij het berichtSoort is het overigens meestal 1, dus dat is het verzoek dat je als eerst moet uitvoeren

**Comment by ghost** [comment]
Als je de mail opent bevat het HTML code met spans en paragraphs, ik zou die er gewoon uit filteren en alleen `<b>, <i>, <em> en <strong>` bewaren en die tekst dikgedrukt en schuingedrukt maken.

**Comment by ghost** [comment]
@Z3r0byte Als je hulp nodig hebt moet je maar aan mij vragen om de HTTP requests uit te voeren aangezien ik nu eindelijk weet hoe dat moet.

**Comment by ghost** [comment]
@Z3r0byte
>Bij het berichtSoort is het overigens meestal 1, dus dat is het verzoek dat je als eerst moet uitvoeren.

Voor de meeste efficiëntie, is wat ik wilde zeggen.

**Comment by ghost** [comment]
Zijn er nog updates?

**Comment by Z3r0byte** [comment]
Hey Stan, de proefwerkweek is net afgelopen, maar ik heb nog wel een aantal PO's die ik af moet maken. Ik hoop dat ik die dit weekend af krijg, zodat ik daarna weer meer aan de app kan werken ;)

**Comment by ghost** [comment]
Oh oké dat is goed! Ik was gewoon even nieuwsgierig of er **ooit** nog iets zou gaan gebeuren

**Comment by Z3r0byte** [comment]
Ik ben voorlopig nog niet van plan om te stoppen met de ontwikkeling, wees maar niet bang hoor ;)

**Comment by ghost** [comment]
Dat is goed nieuws!

**Comment by StanNL** [comment]
Het is maar goed dat het bijna vakantie is! ;)

**Comment by Z3r0byte** [comment]
Inderdaad xD

**Comment by Z3r0byte** [comment]
Gebruik jij eigenlijk een GitHub app?

**Comment by StanNL** [comment]
@Z3r0byte Jazeker! Ik gebruik FastHub vanwege het mooie design! En jij?

**Comment by Z3r0byte** [comment]
Eerst gitskarios, maar die heeft een aantal bugs. Sinds een aantal dagen fasthub, en ik geef je gelijk: het design is erg mooi. Het is helaas alleen niet mogelijk om snel al je repo's te zien.

**Comment by Z3r0byte** [comment]
Oh wacht, ik heb net gevonden hoe je snel je repo's kan bekijken 😅

**Comment by StanNL** [comment]
Oh goed dat je het gevonden hebt in ieder geval haha.

Iets wat ik nog wil toevoegen, trouwens, is het feit dat ik heel af en toe (voornamelijk om te testen) mijn eigen online Github Client gebruik, maar dit is meer een soort klein projectje waar ik af en toe in het weekend aan werk als ik me verveel. Het is op zich best een prima client zonder bugs (denk ik), maar ik heb zegmaar niets gedaan aan het design en het is puur een soort experiment van me waarin ik alleen kijk naar de functionaliteit, en niet naar het uiterlijk. Het werkt trouwens ook alleen op desktop en ik heb geen idee hoe belabberd het er wel niet uit zal zien op mobile haha. 

groetjes vanaf mijn eigen github client ;)
(althans, ik hoop dat dit überhaupt bij jou aankomt, dat is natuurlijk nog maar de vraag)

**Comment by StanNL** [comment]
Zo te zien is de comment toch wel aangekomen! Dat is altijd goed nieuws voor een developer ;)

**Comment by Z3r0byte** [comment]
Gaaf man, een eigen GitHub cliënt!  Kan ik hem ook eens uitproberen? :)

**Comment by StanNL** [comment]
Natuurlijk! Hier is de link: http://s.codepen.io/StanNL/full/yMYEgj/ Let alsjeblieft niet op het design dus, maar vooral ook niet op de 'create new repository' knop die overal is ;) (als je wil commenten en dergelijke, moet je inloggen via de 'log in' knop. Ik dacht ik zeg het nog even ;) ).

**Comment by Z3r0byte** [comment]
De zoekfunctie werkt gewoon goed. Inloggen ging helaas niet vanwege het feit dat ik twee-factor authenticatie aan heb staan. Om het design gemakkelijk te verbeteren kun je misschien een css-framework gebruiken zoals bootstrap of materialize ;)

**Comment by Z3r0byte** [comment]
Dit raakt trouwens wel een beetje off-topic xD misschien handig als we dit via skype of discord ofzo doen ;)

**Comment by StanNL** [comment]
@Z3r0byte Fijn dat de rest wel goed werkt. Over twee-factor: Ik denk niet dat het mogelijk is om dat gemakkelijk te implementeren. Ok denk ook niet dat het het waard is, gezien het feit dat ik eigenlijk de enige ben die het gebruikt en het meer een soort test is.

Over het design: Ja ik was toevallig gisteren bezig met (een poging tot) het maken van een mooiere versie. Ik heb in het verleden wel eens libraries gebruikt zoals MDL of materialize, maar ik doe het toch het liefst allemaal zelf voor de verandering.

Over de off-topic-heid: Ik heb helaas geen andere communicatiemiddelen behalve dit en XDA.

**Comment by Z3r0byte** [comment]
Oh, is ook niet heel erg, aangezien dit een heel populaire repo is fzo ;)

**Comment by StanNL** [comment]
Ik was trouwens al bezig met het design (zoals ik al zei). Als je nieuwsgierig bent kun je het hier vinden: http://s.codepen.io/StanNL/full/RVNWYo/.
Het is zoals je waarschijnlijk wel begrijpt nog een work-in-progress, maar we komen er wel!

**Comment by Z3r0byte** [comment]
Ziet er inderdaad al een stuk beter uit
